pkg/adapter/cloudeventingress/adapter.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudeventingress
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/knative/pkg/cloudevents"
```

Comment:
With the plan to move to using cloud-events sdk, perhaps consider using that from the get go.
https://github.com/knative/eventing-sources/pull/214

Another thought is that with the proposed Broker/Trigger model, this ingress might actually make more sense to live in the Eventing repo? But haven't reviewed all, so perhaps I'm wrong.

Reply:
cloud-events sdk is used.